! # Tutorial Video Walkthrough
! # walkthrough_streamlit_eda_made_easy.diff


- TITLE
+ Exploratory Data Analysis or EDA in Data Science made easy with SWEETVIZ, PANDAS PROFILING and Streamlit
--- See more on the post at https://wp.me/p3Vuhl-38u

- URLs
+ ARTICLE :: https://wp.me/p3Vuhl-38u
+ CODE :: https://bit.ly/3u5eEM0
+ WEBSITE http://flaven.net/

- QUICK DESCRIPTION

--- OBJECTIVE :: How to get an EDA as quickly as possible with the best visualization ever.

--- OUPUT :: EDA Dashboard, based on Streamlit, using the well-known libraries Sweetviz and Pandas Profiling

--- AUDIENCE :: Anyone who want to learn about data-science PO :: "improver", "tinker" or "builder" 

- TEK GUIDELINES
--- Streamlit :: https://streamlit.io/
--- SWEETVIZ :: https://libraries.io/pypi/sweetviz
--- Pandas Profiling :: https://pypi.org/project/pandas-profiling/
--- Sweetviz: Automate Exploratory Data Analysis (EDA) :: https://coderzcolumn.com/tutorials/data-science/sweetviz-automate-exploratory-data-analysis-eda
--- Toy datasets :: https://scikit-learn.org/stable/datasets/toy_dataset.html


- PATH CONSOLE
cd /Users/brunoflaven/Documents/03_git/BlogArticlesExamples/streamlit-sweetviz-pandas-profiling-eda-made-easy/


- CONDA ENVIRONMENT
--- requirement using conda to manage development environment
--- create requirements file for the apps
pip freeze > requirements_eda_made_easy_1.txt
pip install -r requirements_eda_made_easy_1.txt


! CONTENT

- PART_1 TEDIOUS (tedious_manual_eda)
An example with a very basic data exploration to show how time consuming can be the EDA building process where you need to code and extract relevant information. Nevertheless, the advantage of do-it by yourself will provide you a good chance to fully understand all the operation required for a EDA.

--- python 001_tedious_manual_eda.py
--- streamlit run 002_streamlit_tedious_manual_eda.py
--- jupyter notebook then select 001_tedious_manual_eda.ipynb


- PART_2 Pandas Profiling (streamlit_eda_made_easy_pandas_profiling_4)
A great and robust library that do the job one for all. An handy "swiss army knife" to investigate quickly dataset and generate clean report supported with Streamlit.

--- streamlit run 002_streamlit_webapp_pandas_profiling.py

- PART_3 Sweetviz (streamlit_eda_made_easy_sweetviz_3)
All my knowledge came from an article. Again, I strongly invite you to read this article that is the best I have found on the web. I have just extended some elements from this article by giving some information and migrate the all stuff presented into Streamlit framework.

--- streamlit run 009_streamlit_webapp_sweetviz.py
--- streamlit run 010_streamlit_webapp_sweetviz.py

--- https://coderzcolumn.com/tutorials/data-science/sweetviz-automate-exploratory-data-analysis-eda

- PART_4 Class (streamlit_eda_made_easy_1)
On GitHub, you can find different approach to build up an "Exploratory Data Analysis" (EDA in Data Science). This one is inspired by Marina Ramalhete from Brazil that is pretty advanced as is it designed with a Class. I have rework few elements to make it more readable and deliver more contextual information when checking the data.

--- streamlit run streamlit_eda_made_easy_001.py

--- https://github.com/marinaramalhete/Exploratory-Data-Analysis-App


- PART_5 Other (streamlit_eda_made_easy_2)
Few examples gathered from different sources mostly Books or Github, made with Streamlit.
--- cd /Users/brunoflaven/Documents/03_git/BlogArticlesExamples/streamlit-sweetviz-pandas-profiling-eda-made-easy/streamlit_eda_made_easy_2/
--- streamlit run 003_all_streamlit_webapp_for_data_science.py

! EXTRA

- HASHTAGS
An #experimentation and #POC for #EDA #ExploratoryDataAnalysis #DataScience with #SWEETVIZ with #PandasProfiling and for sure #Streamlit https://wp.me/p3Vuhl-38u

- GIT COMMIT
git status
git branch
git add .
git commit -am "updates files make it work for video"
git commit -am "change on several README files"
git push origin master






